ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/CommandTopic.java

```diff
@@ -101,52 +101,72 @@ public Iterable<ConsumerRecord<byte[], byte[]>> getNewCommands(final Duration ti
   }
 
   public List<QueuedCommand> getRestoreCommands(final Duration duration) {
-    final List<QueuedCommand> restoreCommands = Lists.newArrayList();
     if (commandTopicBackup.commandTopicCorruption()) {
       log.warn("Corruption detected. "
           + "Use backup to restore command topic.");
-      return restoreCommands;
+      return Collections.emptyList();
     }
+    return getAllCommandsInCommandTopic(
+        commandConsumer,
+        commandTopicPartition,
+        Optional.of(commandTopicBackup),
+        duration
+    );
+  }
 
-    final long endOffset = getEndOffset();
+  public static List<QueuedCommand> getAllCommandsInCommandTopic(
+      final Consumer<byte[], byte[]> commandConsumer,
+      final TopicPartition topicPartition,
+      final Optional<CommandTopicBackup> backup,
+      final Duration duration
+  ) {
+    final List<QueuedCommand> commands = Lists.newArrayList();
+    final long endOffset = getEndOffsetFromTopicPartition(commandConsumer, topicPartition);
 
     commandConsumer.seekToBeginning(
-        Collections.singletonList(commandTopicPartition));
+        Collections.singletonList(topicPartition));
 
     log.info("Reading prior command records up to offset {}", endOffset);
 
-    while (commandConsumer.position(commandTopicPartition) < endOffset) {
+    while (commandConsumer.position(topicPartition) < endOffset) {
       final ConsumerRecords<byte[], byte[]> records = commandConsumer.poll(duration);
       log.info("Received {} records from command topic restore poll", records.count());
       for (final ConsumerRecord<byte[], byte[]> record : records) {
         try {
-          backupRecord(record);
+          backupRecord(backup, record);
         } catch (final CommandTopicCorruptionException e) {
           log.warn("Backup is out of sync with the current command topic. "
               + "Backups will not work until the previous command topic is "
               + "restored or all backup files are deleted.", e);
-          return restoreCommands;
+          return commands;
         }
 
         if (record.value() == null) {
           continue;
         }
-        restoreCommands.add(
+        commands.add(
             new QueuedCommand(
                 record.key(),
                 record.value(),
                 Optional.empty(),
                 record.offset()));
       }
     }
-    return restoreCommands;
+    return commands;
   }
 
   public long getCommandTopicConsumerPosition() {
     return commandConsumer.position(commandTopicPartition);
   }
 
   public long getEndOffset() {
+    return getEndOffsetFromTopicPartition(commandConsumer, commandTopicPartition);
+  }
+
+  private static long getEndOffsetFromTopicPartition(
```

Comment:
because the CommandTopic object's command consumer is pointed at the bootstrap server from the ksql.command.consumer prefixed configs, whereas we want to get the end offset for the topic on the "old"/original bootstrap.server config, so we need to pass in a different consumer